NUnitTestMarsRover/GridTests.cs

```diff
@@ -0,0 +1,19 @@
+using NUnit.Framework;
+using MarsRover;
+
+namespace NUnitTestMarsRover
+{
+    public class GridTests
+    {
+        [Test]
+        [TestCase(5, 5, 5, 5)]
+        [TestCase(10, 5, 10, 5)]
+        [TestCase(-20, 20, 20, 20)]
+        public void Grid_GivenXYAndGridSize_ReturnsCorrectClassVariableAssignments(int gridX, int gridY, int resultXGrid, int resultYGrid)
+        {
+            var grid = new Grid(gridX, gridY);
+            Assert.That(grid.GridWidth() == resultXGrid &&
```

Comment:
No change requested here, but in the future, I would argue this is testing more than 1 thing.  You're testing both the `GridWidth` and the `GridHeight`.

In theory, you should probably have 2 test functions in this class named something like:

`public void GridWidth_GivenXAndGridSize_ReturnsExpectedWidth(...)`